numpy/core/src/multiarray/descriptor.c

```diff
@@ -589,44 +598,63 @@ _convert_from_list(PyObject *obj, int align)
     if (PyBytes_Check(last_item) && PyBytes_GET_SIZE(last_item) == 0) {
         n = n - 1;
     }
-    /* End ignore code.*/
     if (n == 0) {
+        PyErr_SetString(PyExc_ValueError, "Expected at least one field name");
```

Comment:
This is probably the only testable path, and I'm not sure it's worth the effort.